### PR TITLE
feat: classes for users CRUD operations

### PR DESCRIPTION
**Pull request by olenagerasimova**
Part of #1099 
Added functionality for user's CRUD operations

### CODE REVIEW

File: src/test/java/com/artipie/api/ManageUsersTest.java [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+        final String pass = "xyz";
+        this.users.add(
+            Json.createObjectBuilder().add("type", "plain").add("pass", pass)
+                .add("email", email)
+                .add("groups", Json.createArrayBuilder().add("reader").add("creator").build())
+                .build(),
+            alice
+        );
+        final JsonArray list = this.users.list();
+        final JsonObject nuser = list.stream()
```
Comment (andpopov) [inline src/test/java/com/artipie/api/ManageUsersTest.java:118]: Please add checking of availability of added user
Something like this:
        MatcherAssert.assertThat(
            "Failed to find added user",
            list.stream()
                .anyMatch(usr -> alice.equals(usr.asJsonObject().getString("name"))),
            new IsEqual<>(true)
        );

File: src/test/java/com/artipie/api/ManageUsersTest.java [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+            .filter(usr -> alice.equals(usr.asJsonObject().getString("name")))
+            .findFirst().get().asJsonObject();
+        MatcherAssert.assertThat(
+            "Failed to add user email",
+            nuser.getString("email"),
+            new IsEqual<>(email)
+        );
+        MatcherAssert.assertThat(
+            "Failed to add password",
+            nuser.getString("pass").toString().equals(pass)
```
Comment (andpopov) [inline src/test/java/com/artipie/api/ManageUsersTest.java:128]: Please remove extra 'toString', something like this:
nuser.getString("pass").equals(pass)

File: src/test/java/com/artipie/api/ManageUsersTest.java [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+            new CredsConfigYaml().withUsers("John", "Mark").yaml().toString()
+                .getBytes(StandardCharsets.UTF_8)
+        );
+        this.users.remove("John");
+        MatcherAssert.assertThat(
+            this.users.list().size(),
+            new IsEqual<>(1)
+        );
+    }
+
```
Comment (andpopov) [inline src/test/java/com/artipie/api/ManageUsersTest.java:166]: Please add negative scenario for case removing of non existing user.

**Review by andpopov — CHANGES_REQUESTED** [review]

File: src/main/java/com/artipie/api/ManageUsers.java [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+                    )
+                ).build()
+            ).collect(Collectors.toList())
+        ).orElse(Collections.emptyList()).forEach(builder::add);
+        return builder.build();
+    }
+
+    @Override
+    public void add(final JsonObject info, final String uid) {
+        YamlMappingBuilder builder = Yaml.createYamlMappingBuilder();
```
[the remark this answers is not in the document]
Comment (olenagerasimova) [reply src/main/java/com/artipie/api/ManageUsers.java:75]: This is just an `update` case, why throw an exception here?

File: src/main/java/com/artipie/api/ManageUsers.java [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+                        new String(this.blsto.value(this.key), StandardCharsets.UTF_8)
+                    ).readYamlMapping().yamlMapping("credentials")
+                );
+            } catch (final IOException err) {
+                throw new UncheckedIOException(err);
+            }
+        }
+        return res;
+    }
+}
```
[the remark this answers is not in the document]
Comment (olenagerasimova) [reply src/main/java/com/artipie/api/ManageUsers.java:133]: Update operation is a part of `add` with the case when user already exists

File: src/test/java/com/artipie/api/ManageUsersTest.java (hunk 3 above)
Comment (olenagerasimova) [reply src/test/java/com/artipie/api/ManageUsersTest.java:166]: check line 140 please, there is such test

File: src/main/java/com/artipie/api/ManageUsers.java (hunk 5 above)
[the remark this answers is not in the document]
Comment (andpopov) [reply src/main/java/com/artipie/api/ManageUsers.java:133]: Ok, 'add' will includes 'update' also

File: src/test/java/com/artipie/api/ManageUsersTest.java (hunk 3 above)
Comment (andpopov) [reply src/test/java/com/artipie/api/ManageUsersTest.java:170]: ok, thanks

**Comment by olenagerasimova** [comment]
@andpopov thanks, corrected and answered. Also, I've renamed `add` method into `addAndUpdate` to avoid confusion

**Review by andpopov — APPROVED** [review]
@olenagerasimova thanks!

**Pull request merged**